### PR TITLE
fix(Tag): update filter tags to match updated spec

### LINKED ISSUE #7630: [tag] fix hover and focus for filterable tag
- The current hover and focus states are slightly off for our filterable tag component.
- Since we are also adding a `small` tag variation, we need to make an adjustment to these states so they are consistent across both sizes. 

Artboard Copy 4

------
Artboard Copy 5

### PR DESCRIPTION
**Pull request by emyarod**
Closes #7630
related #7619

This PR updates the filter tag and the new small filter tags to match the updated component spec. Unblocks #7518

#### Testing / Reviewing

Ensure the filter tag variants match the component spec

### CODE REVIEW

**Review by laurenmrice — APPROVED** [review]
Looks great! Thank you 🙌🏻

File: packages/components/src/components/tag/_tag.scss [hunk 1]
```diff
@@ -195,7 +200,7 @@
   .#{$prefix}--tag__close-icon:focus {
     border-radius: 50%;
     outline: none;
-    box-shadow: inset 0 0 0 2px $focus;
+    box-shadow: inset 0 0 0 1px $focus;
```
Comment (tw15egan) [inline packages/components/src/components/tag/_tag.scss:204]: Do we also need to change this on L214?

File: packages/components/src/components/tag/_tag.scss [hunk 2]
```diff
@@ -156,10 +158,13 @@
...
+    display: flex;
     flex-shrink: 0;
-    width: rem(16px);
-    height: rem(16px);
-    margin: 0 0 0 rem(4px);
+    align-items: center;
+    justify-content: center;
+    width: rem(24px);
+    height: rem(24px);
+    margin: 0 0 0 rem(2px);
```
Comment (tw15egan) [inline packages/components/src/components/tag/_tag.scss:167]: Can we add in our transition variables here?

```scss
transition: background-color $duration--fast-01 motion(standard, productive),
      box-shadow $duration--fast-01 motion(standard, productive);
```

File: packages/components/src/components/tag/_tag.scss (hunk 1 above)
Comment (emyarod) [reply packages/components/src/components/tag/_tag.scss:204]: I kept it since it was high contrast specific but I'll reduce it for consistency

Comment (tw15egan) [reply packages/components/src/components/tag/_tag.scss:204]: Yeah just wasn't sure if it was intentional or not, just wanted to bring it up 👍

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by dakahn — APPROVED** [review]

**Pull request merged**